langchain/src/chains/api/api_chain.ts

```diff
@@ -64,16 +70,46 @@ export class APIChain extends BaseChain implements APIChainInput {
   ): Promise<ChainValues> {
     const question: string = values[this.inputKey];
 
-    const api_url = await this.apiRequestChain.predict(
+    const api_json = await this.apiRequestChain.predict(
       { question, api_docs: this.apiDocs },
       runManager?.getChild()
     );
 
-    const res = await fetch(api_url, { headers: this.headers });
+    let api_options;
+    try {
+      api_options = await APIChain.getApiParser().parse(api_json);
```

Comment:
`getAPIParser` can actually just return the output fixing parser directly:

```
      const fixParser = OutputFixingParser.fromLLM(
        this.llm,
        APIChain.getApiParser()
      );
```
The above will automatically use the internal output parser first, then retry with a more focused prompt.

Reply:
Oh should have seen the code. Documentation made me a bit confused. Will fix the documentation as well in another PR.

Reply:
You’re right, definitely could be clearer. Thank you!